app/components/common/CustomModal.tsx

```diff
@@ -49,80 +51,104 @@ const styles = StyleSheet.create({
+  blueLastBtn: {
+    width: widthPercentage(250),
+    height: heightPercentage(44),
+    marginTop: heightPercentage(10),
+    borderRadius: widthPercentage(15),
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: color.mainColor,
+    elevation: 5,
   },
-  secondBtnTitleStyle: {
+  blueLastBtnTitle: {
     fontFamily: 'NotoSansKR-Regular',
     fontSize: fontPercentage(15),
-    fontStyle: 'normal',
     textAlign: 'center',
-    color: '#000000',
+    color: 'white',
   },
 });
 
-interface ModalsProps {
-  mdVisible: boolean;
-  title: string;
-  subtitle?: string;
-  firstButton: Function;
-  secondButton?: Function;
-  firstBtnTitle: string;
-  secondBtnTitle?: string;
-}
-
 function CustomModal({
   mdVisible,
   title,
-  subtitle = null,
-  firstButton,
-  secondButton = null,
-  firstBtnTitle,
-  secondBtnTitle = null,
+  subtitle = '',
+  buttonList,
 }: ModalsProps) {
+  const last = buttonList.pop();
+  const second = buttonList.pop(); // if the array is empty, return undefined
+
   return (
     <Modal animationType="slide" visible={mdVisible} transparent={true}>
       <View style={styles.modalView}>
-        <View style={styles.contentContainer}>
-          <Text style={styles.titleStyle}>{title}</Text>
-          <Text style={styles.subtitleStyle}>{subtitle}</Text>
-        </View>
-        <View style={styles.btnContainer}>
-          <CustomBtn
-            title={firstBtnTitle}
-            onClickListener={firstButton}
-            titleStyle={styles.firstBtnTitleStyle}
-            btnStyle={styles.firstBtnStyle}
-          />
-          {secondButton ? (
+        {second === undefined ? (
+          <View style={styles.oneBtnContent}>
+            <Text style={styles.titleStyle}>{title}</Text>
+            <Text style={styles.subtitleStyle}>{subtitle}</Text>
+          </View>
+        ) : (
+          <View style={styles.content}>
+            <Text style={styles.titleStyle}>{title}</Text>
+            <Text style={styles.subtitleStyle}>{subtitle}</Text>
+          </View>
+        )}
+        {buttonList.map((result, i) => {
+          return (
             <CustomBtn
-              title={secondBtnTitle}
-              onClickListener={secondButton}
-              titleStyle={styles.secondBtnTitleStyle}
-              btnStyle={styles.secondBtnStyle}
+              key={i}
+              title={result.buttonText}
+              onClickListener={result.buttonClickListener}
+              titleStyle={styles.BtnListTitleStyle}
+              btnStyle={styles.BtnListStyle}
             />
-          ) : (
-            <View />
-          )}
-        </View>
+          );
+        })}
+        {second !== undefined ? (
+          <CustomBtn
+            title={second.buttonText}
+            onClickListener={second.buttonClickListener}
+            titleStyle={styles.BtnListTitleStyle}
+            btnStyle={styles.whiteLastBtn}
+          />
+        ) : (
+          <View />
+        )}
```

Comment:
- && 를 써서 불필요한 삼항 연산자를 없애주세요.
```suggestion
        {second !== undefined && (
          <CustomBtn
            title={second.buttonText}
            onClickListener={second.buttonClickListener}
            titleStyle={styles.BtnListTitleStyle}
            btnStyle={styles.whiteLastBtn}
          />
        )}
```